plenum/test/input_validation/test_strict_schema.py

```diff
@@ -0,0 +1,133 @@
+import pytest
+
+from plenum.common.constants import CURRENT_PROTOCOL_VERSION, TXN_TYPE, TARGET_NYM, VERKEY, NYM
+from plenum.common.messages.client_request import ClientMessageValidator
+from plenum.common.messages.fields import NonNegativeNumberField
+from plenum.common.messages.message_base import MessageBase
+from plenum.common.request import SafeRequest
+from plenum.common.types import f, OPERATION
+from plenum.test.input_validation.constants import TEST_TARGET_NYM, TEST_VERKEY_ABBREVIATED
+
+
+class MessageTest(MessageBase):
+    typename = 'MessageTest'
+    schema = (
+        ('a', NonNegativeNumberField()),
+        ('b', NonNegativeNumberField()),
+    )
+
+
+class StrictMessageTest(MessageBase):
+    typename = 'MessageTest'
+    schema = (
+        ('a', NonNegativeNumberField()),
+        ('b', NonNegativeNumberField()),
+    )
+    schema_is_strict = True
+
+
+def test_not_strict_by_default_args():
+    msg = MessageTest(1, 2, 3)
+    assert msg
+    assert msg.a == 1
+    assert msg.b == 2
+
+
+def test_not_strict_by_default_kwargs():
+    kwargs = {'a': 1, 'b': 2, 'c': 3}
+    msg = MessageTest(**kwargs)
+    assert msg
+    assert msg.a == 1
+    assert msg.b == 2
+
+
+def test_strict_with_invalid_args_number():
+    with pytest.raises(ValueError) as excinfo:
+        StrictMessageTest(1, 2, 3)
+    assert ("should be less than or equal to the number of fields "
+            "in schema {}".format(len(MessageTest.schema))) in str(excinfo.value)
+
+
+def test_strict_with_invalid_kwargs_number():
+    kwargs = {'a': 1, 'b': 2, 'c': 3}
+    with pytest.raises(ValueError) as excinfo:
+        StrictMessageTest(**kwargs)
+    assert ("should be less than or equal to the number of fields "
+            "in schema {}".format(len(MessageTest.schema))) in str(excinfo.value)
+
+
+def test_client_req_not_strict_by_default():
+    validator = ClientMessageValidator(operation_schema_is_strict=False)
```

Comment:
If the test is intended for verifying the default behavior then no flag should be given when constructing the validator, right?